### PR TITLE
Docs for archive data migration to berkeley format

### LINKED ISSUE #14857: Archive node migration documentation
Instructions for mainnet archive node migration
1. using the DB dump 
2. performing the migration using the tools developed

### PR DESCRIPTION
**Pull request by dkijania**
Guides for hardfork archive migration:

Closes #14857

### CODE REVIEW

**Comment by barriebyron** [comment]
@dkijania, congratulations on this doc! I made updates and edits directly to the content. See my questions.

You have made an excellent start to the archive migration docs. Do you want this doc published on the docs website https://docs.minaprotocol.com/? 

If so, we must move the PR to the correct docs repo https://github.com/o1-labs/docs2

See the docs README https://github.com/o1-labs/docs2/blob/main/README.md and contributing guidelines https://github.com/o1-labs/docs2/blob/main/CONTRIBUTING.md

If you want to publish in the docs, lmk and I can work with you on those logistics.

File: docs/archive_data_migration_to_berkeley.md [hunk 1]
```diff
@@ -0,0 +1,470 @@
+# Migrating Mainnet Database to Berkeley
+
```
Comment (barriebyron) [inline docs/archive_data_migration_to_berkeley.md:2]: I feel like we can add an overview, a "thanks for your patience" and "we have got you" type of introduction. The doc will benefit from a high level overview of the planning steps, execution steps, and verification steps.

File: docs/archive_data_migration_to_berkeley.md [hunk 2]
```diff
@@ -0,0 +1,470 @@
...
+To run replayer:
+
+```sh
+mina-replayer --archive-uri {db_connection_string} --input-file reference_replayer_input.json --output-file reference_replayer_output.json --checkpoint-interval 100
+```
+
+> :warning: Running a replayer from scratch on a Mainnet database can take up to a couple of days. The recommended best practice is to break the replayer into smaller parts by using the checkpoint capabilities of the replayer.
+
+> :warning: ** You must run replayer from the Mainnet version. You can run it from the Docker image at minaprotocol/mina-archive:1.4.0-c980ba8-bullseye
```
Comment (barriebyron) [inline docs/archive_data_migration_to_berkeley.md:125]:
```suggestion
> :warning: You must run replayer from the Mainnet version. You can run it from the Docker image at minaprotocol/mina-archive:1.4.0-c980ba8-bullseye
```

Comment (barriebyron) [reply docs/archive_data_migration_to_berkeley.md:125]: let's be sure to add active links to the Docker images

File: docs/archive_data_migration_to_berkeley.md [hunk 3]
```diff
@@ -0,0 +1,470 @@
+# Migrating Mainnet Database to Berkeley
+
+Before you start planning the migration of your Mainnet database into the Berkeley version of the Mina network, read this document in entirety.
+
+## Migration apps
+
+Two applications are required to migrate the Mainnet database into the Berkeley version of the Mina network. 
+
+These applications are shipped by MF/O1 Labs:
```
Comment (barriebyron) [inline docs/archive_data_migration_to_berkeley.md:9]: I know this is the intro, but how are these applications shipped? 

(I don't want us to call them "first migration app" and "second migration app" let's call them by name)

File: docs/archive_data_migration_to_berkeley.md [hunk 4]
```diff
@@ -0,0 +1,470 @@
...
+```
+docker pull gcr.io/o1labs-192920/mina-archive:2.0.0rampup8-berkeley-b9e1116-bullseye
+```
+
+## Mainnet Genesis Ledger
+
+## Berkeley database schema files
+
+
+## Migration Steps
```
Comment (barriebyron) [inline docs/archive_data_migration_to_berkeley.md:173]: we actually have several migration steps
1. Plan: read, get the migration apps, make sure you can log in to all databases, and so on
2. Gather info and download the required assets
3. Run migration phase 1
4. Run migration phase 2
5. Verify migration
6. Rinse, repeat (which steps can be repeated?)

File: docs/archive_data_migration_to_berkeley.md [hunk 5]
```diff
@@ -0,0 +1,470 @@
...
+
+   An *archive_balances_migrated* schema is created with the Mainnet archive.
+
+### Validate the Mainnet database
+
+The correct Mainnet database state is crucial for a successful migration. 
+
+Missing blocks or invalid ledger hashes are the most frequent issues when dealing with the Mainnet archive. Although this step is optional, it is strongly recommended to verify the archive condition before you start the migration process.
+
+### Known issues
```
Comment (barriebyron) [inline docs/archive_data_migration_to_berkeley.md:90]: these known issues are for which phase? We have another Known issues on line 403 

Do we want to isolate known issues for each section (part of the migration?) or put all together? 

https://github.com/MinaProtocol/mina/pull/14933/files#diff-677a92f8728b33ec9935394888590b6fe92c6157d3db38b1fda79e798ecc68e0R403

File: docs/archive_data_migration_to_berkeley.md [hunk 6]
```diff
@@ -0,0 +1,470 @@
...
+
+The daemon node unavailability can cause the archive node to miss some of the blocks. 
+
+If you are uploading the missing blocks to Google Cloud, the missing blocks can be reapplied from precomputed blocks () and preserve chain continuity. 
+
+To automatically verify and patch missing blocks, use the [download_missing_blocks.sh]() script. Because the `download-missing-blocks` script uses localhost as the database host, you must run it from within the database host.
+
+1. Install the required `mina-archive-blocks` and `mina-missing-blocks-auditor` scripts that are packed in the `minaprotocol/mina-rosetta:1.4.0-c980ba8-bullseye` Docker image.
+
+2. Export the `BLOCKS_BUCKET:
```
Comment (barriebyron) [inline docs/archive_data_migration_to_berkeley.md:102]:
```suggestion
2. Export the `BLOCKS_BUCKET`:
```

File: docs/archive_data_migration_to_berkeley.md [hunk 7]
```diff
@@ -0,0 +1,470 @@
...
+
+- **berkeley-migration**
+
+  The berkeley-migration application migrates as much data as possible from the Mainnet database and downloads precomputed blocks to get the window density data. 
+  
+  This application runs against the Mainnet and the new database.
+
+- **replayer app in migration mode**
+
+  The replayer app in migration mode replays the transactions in the partially migrated database and populates the `accounts_accessed` and `accounts_created` tables. This application also performs the checks performed by the standard replayer, except that ledger hashes are not checked because the hard fork ledger has greater depth, which results in different hashes. This application runs only against the new database.
```
Comment (barriebyron) [inline docs/archive_data_migration_to_berkeley.md:19]: So, we use the replayer application twice in the migration. 

First: run in migration mode against new database.
Second: Run the replayer application against your existing Mainnet database to verify the blockchain state.

Seems like we can mention here

File: docs/archive_data_migration_to_berkeley.md [hunk 8]
```diff
@@ -0,0 +1,470 @@
...
+To get a fully migrated database:
+
+1. First, run the berkeley-migration application.
+2. Next, run replayer app in migration mode. 
+
+The applications are able to work incrementally so that part of the Mainnet database can be migrated and, as new blocks are added to the Mainnet, the new data in the database can be migrated.
+
+To obtain that incrementality, the berkeley-migration application can look at the migrated database and determine the most recent migrated block. It can continue migrating starting at the next block in the Mainnet data. The replayer app in migration mode can use the checkpoint mechanism already in place for the replayer. A checkpoint file indicates the global slot since genesis for starting the replay and the ledger to use for that replay. The application writes new checkpoint files as it proceeds.
+
+To take advantage of the incrementality, you can run a cron job that migrates a day's worth of data at a time (or some other interval). With the cron job in place, at the time of the actual hard fork, only a small amount of data will need to be migrated.
```
Comment (barriebyron) [inline docs/archive_data_migration_to_berkeley.md:32]: this information is key! Prep now and migrate only a teensy bit later

(let's add to the intro)

**Comment by dkijania** [comment]
Closing this PR in favour to https://github.com/o1-labs/docs2/pull/807

**Pull request closed without merging**